include/input_parser.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef INPUT_PARSER_HPP
+#define INPUT_PARSER_HPP
+
+#include <deal.II/base/parameter_handler.h>
+#include <string>
+#include <iostream>
+#include <fstream> 
+
+namespace WaveEquationProject {
+    using namespace dealii;
+
+    class InputParser {
+    public:
+        InputParser();
+
+        // Funzione principale di parsing
+        void parse_parameters(const std::string &filename);
+
+        // Getters
+        double get_time_step() const;
+        double get_final_time() const;
+        unsigned int get_output_frequency() const;
+        unsigned int get_scenario_id() const;
+        
+        std::string get_solver_type() const; 
+
+        double get_beta() const;
+        double get_gamma() const;
+
+    private:
+        // 'mutable' permette ai metodi const di modificare lo stato interno
+        mutable ParameterHandler prm;
```

Comment:
I assume you declared this as `mutable` so that getter methods of this class could access the parameter values stored inside the handler.

It would probably be better to read the parameter values already inside the `parse_parameters` method and storing those values as class members, so that the getters would only return the value of those members, and `prm` would not need to be declared as `mutable`. This would also be somewhat more efficient, since every time you read a parameter from `ParameterHandler` there's a string-to-something conversion involved.